File: crates/calib-targets/examples/generate_printable.rs
use calib_targets::printable::{write_target_bundle, PrintableTargetDocument};
use std::{env, path::PathBuf};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let spec_path = env::args()
        .nth(1)
        .map(PathBuf::from)
        .unwrap_or_else(|| PathBuf::from("testdata/printable/charuco_a4.json"));
    let out_stem = env::args()
        .nth(2)
        .map(PathBuf::from)
        .unwrap_or_else(|| PathBuf::from("tmpdata/printable/charuco_a4"));

    let doc = PrintableTargetDocument::load_json(&spec_path)?;
    let written = write_target_bundle(&doc, &out_stem)?;
    println!("{}", written.json_path.display());
    println!("{}", written.svg_path.display());
    println!("{}", written.png_path.display());
    Ok(())
}

File: crates/calib-targets-py/examples/generate_printable.py
from __future__ import annotations

import sys
from pathlib import Path

import calib_targets as ct


def main() -> None:
    out_stem = Path(sys.argv[1]) if len(sys.argv) > 1 else Path("tmpdata/printable_charuco")
    doc = ct.charuco_document(
        rows=5,
        cols=7,
        square_size_mm=20.0,
        marker_size_rel=0.75,
        dictionary="DICT_4X4_50",
    )
    written = ct.write_target_bundle(doc, out_stem)
    print(written.json_path)
    print(written.svg_path)
    print(written.png_path)


if __name__ == "__main__":
    main()
